Sources/SwiftParser/Lexer.swift

```diff
@@ -51,6 +51,12 @@ public struct Lexer {
       return self.flags.contains(.isMultilineStringLiteral)
     }
 
+    var isEditorPlaceholder: Bool {
```

Comment:
`parseIdentifierExpression` has this exact check as well. Could use this there.

There's also a bunch of others (differing slightly):
  - `SyntaxClassifier.contextFreeClassification`
  - `Lexer.lexOperatorIdentifier` and `Lexer.tryLexEditorPlaceholder`
  - `SyntaxText.containsPlaceholderEnd`

Would be nice not to have so many 😅

Reply:
The problem is that all of these work on slightly different types/have slightly different semantics:

- This implementation: Full text match on `Lexer.Lexeme`
- `SyntaxClassifier.contextFreeClassification`: Full text match on `RawTokenKind` + `SyntaxText`
- `Lexer.lexOperatorIdentifier` and `Lexer.tryLexEditorPlaceholder`: Substring match on `UInt8`
- `SyntaxText.containsPlaceholderEnd`: Implementation detail of the above

I tried unifying them but it isn’t really possible 😬

Reply:
Could at least unify `parseIdentifierExpression`

Reply:
What would you like to unify `parseIdentifierExpression` with?

Reply:
This one. They both use `SyntaxText` -
`parseIdentifierExpression`:
```
name.tokenText.hasPrefix("<#") && name.tokenText.hasSuffix("#>")
```
`isEditorPlaceholder`:
```
self.tokenText.starts(with: SyntaxText("<#")) &&
       self.tokenText.hasSuffix(SyntaxText("#>"))
```

`SyntaxText.isEditorPlaceholder`? 🤷

Reply:
Ah, of course. Will do.